### DISCUSSION TITLE
Different formatting rules for tsconfig.json and tsconfig.node.json when using Vite

### DISCUSSION THREAD

**Discussion by lewiswolf** [prettier/prettier#15952] [Q&A]
In the most recent update of prettier, tsconfig.json files are now formatted with trailing commas when using `"trailingComma": "all"`, but this change does not effect tsconfig.node.json, which does not have trailing commas. This file is a part of the setup for a  Vite environment. I like trailing commas! But I would like the two files to be stylised consistently. :) Thanks!

**Comment by starkovsky** [comment]
Fast fix for this behavior set parser to json for all json files or tsconfig only
```json
	"overrides": [
		{
			"files": "*.json",
			"options": {
				"parser": "json"
			}
		}
	]
```

**Reply by electrovir** [reply]
Or just lock in your Prettier version until it's fixed

**Comment by lewiswolf** [comment]
curious why tsconfig.json is a special case in the first place? why is it not auto-formatted like every other json file?

**Comment by Punie** [comment]
This is extremely annoying for a lot of reasons:
1. it's a breaking change as part of a patch version (although semver in the JS ecosystem has always been a joke to most people)
2. the proposed solution with `overrides` does not seem to work as is if I have a monorepo with a `@project/prettier-config` package with the override in it. I have to somehow repeat this in every package of my project instead of just having `"prettier": "@project/prettier-config"` in my `package.json`
3. as mentioned [here](https://github.com/prettier/prettier/issues/15942#issuecomment-1897670777), Microsoft [explicitly discourages trailing commas in `jsonc`](https://code.visualstudio.com/docs/languages/json#_json-with-comments)
4. if I want my `tsconfig.json` to be handled as `.jsonc` I should be able to either opt-in to it with prettier overrides **OR** explicitly having the `.jsonc` extension
5. I respectfully disagree with the statement offered [here](https://github.com/prettier/prettier/issues/15942#issuecomment-1894985961) but I also find it kinda rude and dismissive

While I generally am 100% in agreement with whatever `prettier` does and I usually never change any of it's default (I've always been on board with [this](https://prettier.io/docs/en/option-philosophy) for example), I do believe there is something wrong in this particular case.
Trailing commas can and **DO** pose actual issues in a lot of cases within `.jsonc` files, they are not considered idiomatic to the file format and as such they should be treated differently (or at the very least, automatic inference for file extensions not matching `.jsonc` should be opt-in).

**Reply by ghiscoding** [reply]
related to issue #15956 and #15945
